simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/component/StationComponent.java

```diff
@@ -94,4 +96,23 @@ public SetVariableValidationResult validate(SetVariableDatum setVariableDatum) {
         return new SetVariableValidationResult(setVariableDatum, validationResult);
     }
 
+    /**
+     * Generates report data for all variable in the component
+     *
+     * @param onlyMutableVariables if true, returns only those variables that can be set by the operator
+     * @return list of {@link ReportDatum}
+     */
+    public List<ReportDatum> generateReportData(boolean onlyMutableVariables) {
+        List<ReportDatum> reportData = new ArrayList<>();
+        variableAccessors.values().forEach(accessor -> {
+            if (shouldGenerateReportDataForVariable(onlyMutableVariables, accessor.isMutable())) {
```

Comment:
Consider using Stream API. Such as: 
`variableAccessors.values().stream().filter(VariableAccessor::isMutable).flatMap(variableAccessor -> variableAccessor.generateReportData(getComponentName()).stream()).collect(toList());`

Reply:
That's only for the case where onlyMutable == true, you will stave have an if (onlyMutable) { that stream code} else { ... }

I think what we have now is more readable, unless you know of a way to integrate onlyMutable into that lambda.

Reply:
this expression can be improved -- onlyMutableVariables && isMutableVariable || !onlyMutableVariables;

Reply:
I used an online boolean logic calculator 👍 

Changed it to return !onlyMutableVariables || isMutableVariable;